src/components/Feed/MoviesFeed.tsx

```diff
@@ -93,22 +93,35 @@ const MoviesFeed: React.FC = () => {
         </CardContent>
       </Card>
 
-      {Array.from(movieIds).map((id) => (
-        <div
-          key={id}
-          onClick={() => navigate(`${id}`)}
-          style={{ cursor: "pointer" }}
-        >
-          <MovieCard imdbId={id} />
-        </div>
-      ))}
+      {loading && movieIds.size === 0 ? (
+        <Box display="flex" justifyContent="center" py={8}>
+          <CircularProgress />
+        </Box>
+      ) : (
+        <Box>
+          {Array.from(movieIds).map((id) => (
+            <div
+              key={id}
+              onClick={() => navigate(`${id}`)}
+              style={{ cursor: "pointer" }}
+            >
+              <MovieCard imdbId={id} />
+            </div>
+          ))}
+        </Box>
+      )}
 
-      {hasMore && (
-        <Card sx={{ mt: 2, cursor: "pointer" }} onClick={fetchBatch}>
-          <CardContent>
-            <Typography align="center">Load More</Typography>
-          </CardContent>
-        </Card>
+      {hasMore && movieIds.size > 0 && (
```

Comment:
our has more logic, is stupid i think, it only waits for enough movies for the batch size, which shouldn;t be the case, i think load more button should always exist, and should be a "since" filter and we should dial it back to earlier dates. Right now I know movies and review exist but the load more button vanishes, which is just wrong UI.